### ISSUE TITLE
Some format can not parse, ngnix

### ISSUE THREAD

**Issue by hiql** [nielsbasjes/logparser#227]
Hi:

I have tried this cool lib, and found some issues, if the “request” column is empty or has some illegal characters, the parser will throw an exception, sometimes we want to keep the successful columns and ignore the failed columns(or return the original text).
How can I solve the problem?

not ok:
```
94.232.44.112 - - [28/Feb/2021:03:54:40 +0800] "x00Cookie: mstshash=Administr" 400 157 "-" "-" "-"
207.154.195.167 - - [01/Mar/2021:03:25:25 +0800] "" 400 0 "-" "-" "-"

```
very ok:
```
183.136.225.42 - - [28/Feb/2021:20:06:29 +0800] "GET /logo.jpg HTTP/1.1" 200 1394 "-" "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36 QIHU 360SE" "-"
```

my nginx format:
`$remote_addr - $remote_user [$time_local] "$request" $status $body_bytes_sent "$http_referer" "$http_user_agent"`

**Comment by nielsbasjes** [comment]
Looking at your logformat and the provided lines I see that all of the lines have an extra "-" at the end. I'll have a look at what is happening here and how you can work keep the broken columns.

**Comment by hiql** [comment]
maybe the column "$request" needs some special format?

`"x00Cookie: mstshash=Administr"`

I think the "x00" could be the problem?

**Comment by nielsbasjes** [comment]
No, that is not the problem. I'm working on a a fix.

**Comment by nielsbasjes** [comment]
I have released version 5.9.0 which fixes the problem.
